### PR TITLE
Add the wait for subscription payment step

### PR DESCRIPTION
**Pull request by richardW8k**
This adds a new step which pauses the workflow until a subscription event occurs. The step status options are used to define what should happen for each event:

- Upcoming Payment (invoice.upcoming)
- Paid (invoice.payment_succeeded)
- Failed (invoice.payment_failed)
- Cancelled (customer.subscription.deleted)

## Testing Instructions
- Go to https://dashboard.stripe.com/settings/billing/automatic and set the choice for the **Upcoming renewal events** setting to the shortest period
- Create a Stripe subscription feed with a daily billing cycle
- Create an approval step for each of the above events
- Create a wait for subscription payment step assigning the steps to the status
- Move the wait step so it is the first step
- Submit form
- When the invoice.payment_succeeded event occurs find the workflow moves to the specified step
- Go to https://dashboard.stripe.com/test/customers
- Delete the payment method from the customer
- Add 4000000000000341 as the new payment method
- Return to site
- Find entry and send to the wait step
- When the invoice.payment_failed event occurs find the workflow moves to the specified step
- Find entry and send to the wait step
- Cancel subscription
- Find the workflow moves to the specified step
- Edit stripe feed and change billing cycle to 4 days
- Submit form
- When the invoice.upcoming event occurs find the workflow moves to the specified step

### CODE REVIEW

File: class-stripe.php [hunk 1]
```diff
@@ -289,6 +307,43 @@ public function action_post_add_subscription_payment( $entry ) {
...
+		     || ! gf_stripe()->is_payment_gateway( $entry['id'] )
+		) {
+			return;
+		}
+
+		$api  = new Gravity_Flow_API( $entry['form_id'] );
+		$step = $api->get_current_step( $entry );
+
+		if ( $step && $step->get_type() === 'stripe_wait_subscription_payment' ) {
+			$step->log_debug( sprintf( '%s(): Processing action %s for entry #%d.', __METHOD__, $action_type, $entry['id'] ) );
```
Comment (yoren) [inline class-stripe.php:334]: Will it be helpful if we can add an entry note when the custom `upcoming_subscription_payment` event type is triggered? I couldn't tell if that's something GF payment add-on framework already handle automatically.

File: includes/steps/class-step-stripe-wait-subscription-payment.php [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	 *
+	 * @since 1.4
+	 *
+	 * @return array
+	 */
+	public function get_status_config() {
+		return array(
+			array(
+				'status'                    => 'upcoming',
+				'status_label'              => __( 'Upcoming Payment', 'gravityflowstripe' ),
```
Comment (yoren) [inline includes/steps/class-step-stripe-wait-subscription-payment.php:74]: should we use `esc_html__()` instead of `__()`?

**Comment by yoren** [comment]
I have confirmed all 4 events can trigger the Wait step.

File: class-stripe.php (hunk 1 above)
Comment (richardW8k) [reply class-stripe.php:334]: As it's a custom event the framework isn't going to add a note. I don't think it needs a dedicated note as we aren't updating the entry subsctiption status. Also, if the selected next step sends a notification that would add a note for the notification sending result.

**Pull request merged**